user/incus/template.py

```diff
@@ -0,0 +1,81 @@
+pkgname = "incus"
+pkgver = "6.4.0"
+pkgrel = 10
+build_style = "go"
+make_build_args = ["./..."]
+hostmakedepends = [
+    "automake",
+    "libtool",
+    "bash",
+    "git",
+    "go",
+    "go-md2man",
+    "less",
+    "pkgconf",
+]
+makedepends = [
+    "acl-devel",
+    "cowsql-devel",
+    "libseccomp-devel",
+    "libcap-devel",
+    "linux-headers",
+    "libuv-devel",
+    "lxc-devel",
+    "raft-devel",
+    "sqlite-devel",
+    "udev-devel"
+]
+depends = [
+    "libvirt",
+    "acl",
+    "acl-progs",
+    "lxc",
+    "rsync",
+    "squashfs-tools",
+    "xz",
+    "dnsmasq",
+    "iptables",
+    "attr-progs",
+    "util-linux"
+]
+go_build_tags=[ "libsqlite3" ]
+pkgdesc = "Powerful system container and virtual machine manager"
+maintainer = "tj <tjheeta@gmail.com>"
+license = "Apache-2.0"
+url = "https://github.com/lxc/incus"
+source = f"{url}/archive/v{pkgver}.tar.gz"
+sha256 = "c6ad8260a4954b0e4f77de66220d5450290b57858f8c16ee83f12996eea86db4"
+# void ignores TestConvertNetworkConfig (https://github.com/void-linux/void-packages/blob/master/srcpkgs/incus/template)
+# go test -v -tags libsqlite3 -skip TestConvertNetworkConfig ./...
+options = ["!check"]
+
+
+def post_build(self):
+    # Build the agent statically
+    self.env["CGO_ENABLED"] = "0"
+    self.env["CGO_LDFLAGS"] = "-extldflags=-static"
+    self.go_build_tags = ["agent", "netgo", "osusergo"]
+    self.make_build_args = ["-buildmode=default", "./..."]
```

Comment:
buildmode should be pie still

Reply:
hm so

- the extldflags don't do something without also -linkmode=external
- go itself with its own linker seems to not be able to output a static pie executable
- so, either you need what you already put (and can remove the ldflags), or you can cgo_enabled=1 cgo_ldflags=-static-pie and remove the -buildmode. i don't know which is better, but that at least gives a static pie instead

Reply:
Just tried

```
    self.env["CGO_ENABLED"] = "1"
    self.env["CGO_LDFLAGS"] = "-static-pie"
    self.go_build_tags = ["agent", "netgo", "osusergo"]
    self.make_build_args = ["./..."]
    self.golang.build(wrksrc="cmd/incus-agent")
```

Gives the error:
```
/usr/lib/go/pkg/tool/linux_amd64/link: running clang failed: exit status 1
/usr/bin/clang -m64 -Wl,-z,relro -pie -o $WORK/b001/exe/a.out -Wl,--export-dynamic-symbol=_cgo_panic -Wl,--export-dynamic-symbol=_cgo_topofstack -Wl,--export-dynamic-symbol=crosscall2 -Qunused-arguments -Wl,--compress-debug-sections=zlib /tmp/go-link-1580340833/go.o /tmp/go-link-1580340833/000000.o /tmp/go-link-1580340833/000001.o /tmp/go-link-1580340833/000002.o /tmp/go-link-1580340833/000003.o /tmp/go-link-1580340833/000004.o /tmp/go-link-1580340833/000005.o /tmp/go-link-1580340833/000006.o /tmp/go-link-1580340833/000007.o /tmp/go-link-1580340833/000008.o /tmp/go-link-1580340833/000009.o /tmp/go-link-1580340833/000010.o /tmp/go-link-1580340833/000011.o /tmp/go-link-1580340833/000012.o /tmp/go-link-1580340833/000013.o /tmp/go-link-1580340833/000014.o /tmp/go-link-1580340833/000015.o /tmp/go-link-1580340833/000016.o /tmp/go-link-1580340833/000017.o -static-pie -lutil -lpthread -static-pie -lpthread
ld: error: unable to find library -l:libunwind.a
ld: error: unable to find library -latomic
ld: error: unable to find library -lc
clang: error: linker command failed with exit code 1 (use -v to see invocation)

```

Looks like it's still trying to link to musl and others. Going to try to put the static versions as makedepends.

Reply:
Hmmm...I'm not sure if this worked or not.
```
$ ldd /usr/bin/incus-agent 
        ldd (0x70c2db22a000)

```

Reply:
From irc, have to use file to verify that pie binaries are built statically. 

```
file /usr/bin/incus-agent
/usr/bin/incus-agent: ELF 64-bit LSB pie executable, x86-64, version 1 (SYSV), static-pie linked, Go BuildID=01tUkAstqv9ow_1cyjgs/NIM2E9c867DsBpZK4JBg/eTRB6Ypl35xXoqgWjeSM/_tNAfH8EX9KYB8VIpqvp, BuildID[xxHash]=a0f2727ea2699ca7, stripped
```